specs/_features/eip7732/beacon-chain.md

```diff
@@ -0,0 +1,796 @@
+- [Beacon chain state transition function](#beacon-chain-state-transition-function)
+  - [Block processing](#block-processing)
+    - [Withdrawals](#withdrawals)
+      - [Modified `process_withdrawals`](#modified-process_withdrawals)
+    - [Execution payload header](#execution-payload-header)
+      - [New `verify_execution_payload_header_signature`](#new-verify_execution_payload_header_signature)
+      - [New `process_execution_payload_header`](#new-process_execution_payload_header)
+    - [Operations](#operations)
+      - [Modified `process_operations`](#modified-process_operations)
+      - [Payload Attestations](#payload-attestations)
+        - [`process_payload_attestation`](#process_payload_attestation)
+    - [Modified `process_execution_payload`](#modified-process_execution_payload)
+      - [New `verify_execution_payload_envelope_signature`](#new-verify_execution_payload_envelope_signature)
+    - [Modified `is_merge_transition_complete`](#modified-is_merge_transition_complete)
+    - [Modified `validate_merge_block`](#modified-validate_merge_block)
+- [Testing](#testing)
+  - [Modified `initialize_beacon_state_from_eth1`](#modified-initialize_beacon_state_from_eth1)
+  - [Modified `is_merge_transition_complete`](#modified-is_merge_transition_complete-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification of the enshrined proposer builder separation feature. 
+
+*Note:* This specification is built upon [Electra](../../electra/beacon-chain.md) and is under active development.
+
+This feature adds new staked consensus participants called *Builders* and new honest validators duties called *payload timeliness attestations*. The slot is divided in **four** intervals. Honest validators gather *signed bids* (a `SignedExecutionPayloadHeader`) from builders and submit their consensus blocks (a `SignedBeaconBlock`) including these bids at the beginning of the slot. At the start of the second interval, honest validators submit attestations just as they do previous to this feature). At the  start of the third interval, aggregators aggregate these attestations and the builder broadcasts either a full payload or a message indicating that they are withholding the payload (a `SignedExecutionPayloadEnvelope`). At the start of the fourth interval, some validators selected to be members of the new **Payload Timeliness Committee** (PTC) attest to the presence and timeliness of the builder's payload.
+
+At any given slot, the status of the blockchain's head may be either 
+- A block from a previous slot (e.g. the current slot's proposer did not submit its block). 
+- An *empty* block from the current slot (e.g. the proposer submitted a timely block, but the builder did not reveal the payload on time). 
+- A full block for the current slot (both the proposer and the builder revealed on time). 
```

Comment:
Maybe `present`/`absent` would be better names than `full`/`empty`. 
And, using `execution block` explicitly in the spec to avoid ambiguity with the beacon block.